### PR TITLE
Add `authenticating_proxy_ca` to `cce_cluster_v3`

### LINKED ISSUE #622: r/cce_node_v3 doesn't fit doc
Our current implementation of `r/opentelekomcloud_cce_node_v3` doesn't fit to doc in [HelpCenter](https://docs.otc.t-systems.com/api2/cce/cce_02_0242.html)

### LINKED ISSUE #726: Add Support of CA parameter to data_source_opentelekomcloud_cce_cluster_v3
# Description of the new resource/data-source

Terraform currently does not allow to provide a certificate for authentication against the cluster in the `data_source_opentelekomcloud_cce_cluster_v3 ` resource.
 
The parameter used in the api is `ca` see.
https://docs.otc.t-systems.com/en-us/api2/cce/cce_02_0236.html#cce_02_0236__table17313161473013

Can you add this parameter to the CCE cluster resource?
https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/blob/master/opentelekomcloud/data_source_opentelekomcloud_cce_cluster_v3.go

Best Regards 
Martin

### PR DESCRIPTION
**Pull request by outcatcher**
## Summary of the Pull Request
Resolve #726
Resolve #622

Refactor `cce_cluster_v3` code

## PR Checklist

* [x] Refers to: #726
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

#### Resource:
```
=== RUN   TestAccCCEClusterV3_proxyAuth
--- PASS: TestAccCCEClusterV3_proxyAuth (1571.04s)
PASS
```

#### Data Source:
```
=== RUN   TestAccCCEClusterV3DataSource_basic
--- PASS: TestAccCCEClusterV3DataSource_basic (1567.92s)
PASS
```

### CODE REVIEW

**Comment by outcatcher** [comment]
recheck

**Comment by outcatcher** [comment]
recheck

File: opentelekomcloud/data_source_opentelekomcloud_cce_cluster_v3.go [hunk 1]
```diff
@@ -63,6 +64,10 @@ func dataSourceCCEClusterV3() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"authentication_mode": {
```
Comment (anton-sidelnikov) [inline opentelekomcloud/data_source_opentelekomcloud_cce_cluster_v3.go:67]: in doc this is optional value

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: opentelekomcloud/data_source_opentelekomcloud_cce_cluster_v3.go (hunk 1 above)
Comment (outcatcher) [reply opentelekomcloud/data_source_opentelekomcloud_cce_cluster_v3.go:67]: `Computed` without `Required` or `Optional` means that field can't be set by the user at all. For data source such fields are not part of `ListOpts` but returned in the `Get` response

**Review by anton-sidelnikov — APPROVED** [review]

**Review by lego963 — APPROVED** [review]

**Comment by lego963** [comment]
It refers to #622 and I suppose, that could be closed after merge

**Pull request merged**